### ISSUE TITLE
Wrong error on `Callable` with only one argument

### ISSUE THREAD

**Issue by eliegoudout** [python/mypy#14607]
Hello,

On a project, using python3.10 and `mypy 0.991`, I got the following error, which I think should not be raised:
```
<file>: error: The first argument to Callable must be a list of types, parameter specification, or "..."  [valid-type]
```
But I in fact checked the following in python3.10:
```pycon
>>> Callable[int, str]
typing.Callable[[int], str]
```
which makes me think that python manages the special case of a **callable with only one argument**, contrary to what Mypy expects.

The weird thing though is that I can't find any documentation of this behaviour from python...

**Comment by AlexWaygood** [comment]
Mypy is behaving as designed here. The [CPython typing docs state](https://docs.python.org/3/library/typing.html#typing.Callable):

> The subscription syntax must always be used with exactly two values: the argument list and the return type. The argument list must be a list of types or an ellipsis; the return type must be a single type.

`int` is neither a list of types nor an ellipsis. The fact that the CPython runtime does not enforce this rule has no bearing on what the typing spec officially allows. There are many things that do not cause exceptions to be raised at runtime which are nonetheless disallowed by mypy.

**Comment by JelleZijlstra** [comment]
The Python runtime is sometimes more permissive than necessary. I'd have to look up whether that particular variant should be a runtime error, but mypy is definitely in the right about giving an error here.
